### ISSUE TITLE
3 is a 9

### ISSUE THREAD

**Issue by moeskerv** [jomjol/AI-on-the-edge-device#129]
At first: This is an awesome project! I have it running for some days now and it works great. I was looking for an solution for water meter already for a longer time and did not want to use a proximity sensor for it due to its incremental readout nature. This solution here is perfect! It always returns the current complete reading.

One thing I noticed is that for some days now a 3 in the last digital ROI is read as a 9.

![Screenshot 2021-02-28 090645](https://user-images.githubusercontent.com/15142022/109413074-eeefe600-79ab-11eb-991b-ae176158c39c.png)
![Screenshot 2021-02-28 090730](https://user-images.githubusercontent.com/15142022/109413076-f0b9a980-79ab-11eb-99e0-21ec984ee74c.png)

I could fix this issue already by increasing the ROI size a bit:

![Screenshot 2021-02-28 095520](https://user-images.githubusercontent.com/15142022/109413098-09c25a80-79ac-11eb-8103-f96577e6868c.png)

So to sum it up: Might not be an issue at all, but might be of interest if someone has a similar issue.
If I can provide more helpful data on this, I'm happy to help.

Thanks for this great and awesome project!
I built my own adapter and tubus in order to print it w/o supports and in less pieces, also I use one of the ESP32-CAM-MB boards, with bring an USB daughter board so no soldering is required. I will post this as a remix on thingiverse.

BR,
Volker

**Comment by moeskerv** [comment]
Oh oh. Might be yet an issue with my setup:

Again the 3 has become a 9. Also with the increases ROI the issue returns after some readings. 

![image](https://user-images.githubusercontent.com/15142022/109413485-f31d0300-79ad-11eb-9f60-45d125cf08fe.png)

Any advice would be appreciated.

Thanks,
Volker

**Comment by moeskerv** [comment]
All good again in the next cycle:

![image](https://user-images.githubusercontent.com/15142022/109413553-57d85d80-79ae-11eb-939a-9709ec6c39e0.png)

I will further keep an eye on this.

**Comment by jomjol** [comment]
Hi Volker,

you could collect some of the false identified images and send them to me for teaching. I will update the neural netework file from time to time (see just yesterday).

Best regards,
  jomjol

**Comment by moeskerv** [comment]
Hi jomjol

sure I can do that. How should I collect these pictures? Will screenshots like above be ok?

BR,
Volker

**Comment by jomjol** [comment]
Hi Voker,

no screenshots, please! Turn on the log files for the digits and you will find already cut out ROIs in the log directory. Please collect 2-3 typicall images for each digit (0, 1, 2, ... 9) and post them, e.g. here in a zip file.

Thanks,
  jomjol

**Comment by moeskerv** [comment]
Thanks! I will start collection and provide some data about my meter in a few days.

**Comment by jomjol** [comment]
Let me know, when you have new numbers. Until then I will close this issue, to keep the overview on running tasks on my side

**Comment by moeskerv** [comment]
Here are the numbers. Took a while until a full cycle was done (10k Liters). A also added numbers for the gas meter I built.

[gas_watermeter_digits.zip](https://github.com/jomjol/AI-on-the-edge-device/files/6191880/gas_watermeter_digits.zip)

**Comment by jomjol** [comment]
I've teached them in the following new tflite. Just unzip it and test it:
[dig0820s2q.zip](https://github.com/jomjol/AI-on-the-edge-device/files/6192218/dig0820s2q.zip)

Let me know, if it works, then I will take it over in the next release version.

**Comment by moeskerv** [comment]
Unzipped, uploaded and configured in the config.ini. Will check and give a feedback.

**Comment by moeskerv** [comment]
After around 24h I can confirm that the digit recognition seems to work quite well on both meters! No issues with the 3 any more.

**Comment by jomjol** [comment]
Great to hear that!
